test/hw3/test_task1.py

```diff
@@ -0,0 +1,18 @@
+from unittest.mock import patch
+
+from All_home_works.hw3.task1 import cache
+
+
+@cache(times=2)
+def f():
+    return input("? ")
+
+
+@patch("hw3.test_task1.f", return_value=2)
+def test__cache(f):
+    function = cache(3)(f)
+    function()
+    function()
+    function()
+    function()
+    assert 1 == len(f.mock_calls)
```

Comment:
You intrigued me. You call function n times and stopped on a very interesting place.   What will happen next?

Reply:
lets use: `asset_called_once_with`

Reply:
You intrigued me. You call function n times and stopped on a very interesting place. What will happen next?

Function "f" will call one more time. As counter in decorator became zero. And len(f.mock_call) will become 2

Reply:
Please use quote when quoting 

![image](https://user-images.githubusercontent.com/171597/116477744-b4100e80-a885-11eb-9dd9-3aaec3839dc0.png)

Reply:
> Function "f" will call one more time. As counter in decorator became zero. And len(f.mock_call) will become 2

Can you make a solid prove?

Reply:
![Снимок экрана 2021-04-29 в 14 38 13](https://user-images.githubusercontent.com/70700249/116545035-9203a400-a8f8-11eb-93bd-87fd85af6d78.png)

Reply:
Ok. I understood this task differently. Is it needed to remake?
An example from the description not very useful. It will be good if they will use your example, it's more informative.

Reply:
> Ok. I understood this task differently. Is it needed to remake?

Yes. 

> An example from the description not very useful. 

This often happens in real life.  

The situation when you try, fail, think, try fail ....  and finally get the right unswear is very cool for getting experience.  
So having a bit vague task description is not a problem.